int32 encode_data_left
int32 encode_data_right
